File: app/src/main/java/com/savvasdalkitsis/gameframe/widget/view/MenuWidgetProvider.java
package com.savvasdalkitsis.gameframe.widget.view;

import com.savvasdalkitsis.gameframe.R;

public class MenuWidgetProvider extends ClickableWidgetProvider {

    @Override
    protected int layoutResId() {
        return R.layout.widget_menu;
    }

    @Override
    protected void onClick() {
        presenter.menu();
    }
}

File: app/src/main/java/com/savvasdalkitsis/gameframe/widget/view/MenuWidgetProvider.kt
package com.savvasdalkitsis.gameframe.widget.view

import com.savvasdalkitsis.gameframe.R

class MenuWidgetProvider : ClickableWidgetProvider() {

    override fun layoutResId() = R.layout.widget_menu

    override fun onClick() {
        presenter.menu()
    }
}
